src/arrayMethodJoin.js

```diff
@@ -4,8 +4,34 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let strJoin = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    let fixedSeparator = separator;
+
+    if (separator === null) {
+      fixedSeparator = 'null';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      let element = this[i];
+
+      if (element === undefined || element === null) {
+        element = '';
+      }
+
+      strJoin += element + fixedSeparator;
```

Comment:
let's improve it a bit, instead of introducing `let` variable, you can invert your condition and just add element beforehand, like:
```suggestion
      if (condition where you SHOULD add element) {
        // here you add element to your string
      }

      // here you add separator to it
```